code/datums/radiation_wave.dm

```diff
@@ -47,9 +49,8 @@
 	if(strength<RAD_BACKGROUND_RADIATION)
 		qdel(src)
 		return
-
-	radiate(atoms, FLOOR(strength, 1))
-
+	var/last_radiation_amount = radiate(atoms, FLOOR(min(strength,remaining_contam), 1))
```

Comment:
You should still be radiating things with the full force just don't contaminate, add a new argument to the radiate proc

Reply:
done